store/sqlstore/preference_store.go

```diff
@@ -54,31 +54,25 @@ func (s SqlPreferenceStore) DeleteUnusedFeatures() {
 	s.GetMaster().Exec(sql, queryParams)
 }
 
-func (s SqlPreferenceStore) Save(preferences *model.Preferences) store.StoreChannel {
-	return store.Do(func(result *store.StoreResult) {
-		// wrap in a transaction so that if one fails, everything fails
-		transaction, err := s.GetMaster().Begin()
-		if err != nil {
-			result.Err = model.NewAppError("SqlPreferenceStore.Save", "store.sql_preference.save.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
-		} else {
-			defer finalizeTransaction(transaction)
-			for _, preference := range *preferences {
-				if upsertResult := s.save(transaction, &preference); upsertResult.Err != nil {
-					*result = upsertResult
-					break
-				}
-			}
+func (s SqlPreferenceStore) Save(preferences *model.Preferences) (int, *model.AppError) {
```

Comment:
@jespino Not blocking for this PR, but I'm wondering why this returns an `int`. This is used in the EE code? At least in TE it's not.

Reply:
Nope, that code is used only in TE, probably we can remove that int return. But I would prefer to do it in another commit because this one is fully reviewed. @IshankGulati do you want to do it in another PR?

Reply:
Yeah sure. I'd raise another PR for that.